dtfs-central-api/src/repositories/tfm-facilities-repo/tfm-facilities.repo.test.ts

```diff
@@ -0,0 +1,167 @@
+import { ObjectId } from 'mongodb';
+import { ParsedTfmFacility, RawTfmFacility, TfmFacilitiesRepo } from './tfm-facilities.repo';
+
+describe('tfm-facilities-repo', () => {
+  describe('TfmFacilitiesRepo.validateAndParseFindOneResult', () => {
+    const aValidTfmFacilityDocumentFacilitySnapshotResult = () => ({
+      _id: new ObjectId(),
+      dealId: new ObjectId(),
+      type: 'A type',
+      hasBeenIssued: true,
+      name: 'A name',
+      shouldCoverStartOnSubmission: true,
+      coverStartDate: null,
+      coverEndDate: null,
+      issueDate: null,
+      monthsOfCover: 12,
+      details: [],
+      detailsOther: '',
+      value: 9000,
+      coverPercentage: 12,
+      interestPercentage: 6,
+      ukefFacilityId: '12345678',
+      dayCountBasis: 365,
+    });
+
+    const aValidTfmFacilityDocument = () => ({
+      _id: new ObjectId(),
+      facilitySnapshot: aValidTfmFacilityDocumentFacilitySnapshotResult(),
+    });
+
+    it('throws an error when the input is an empty object', () => {
+      // Arrange
+      const input = {} as unknown as RawTfmFacility;
+
+      // Act / Assert
+      expect(() => TfmFacilitiesRepo.validateAndParseFindOneResult(input)).toThrow();
```

Comment:
FWIW we can make sure we're not getting a different error here
```suggestion
      expect(() => TfmFacilitiesRepo.validateAndParseFindOneResult(input)).toThrow(ZodError);
```